js/assessment.js

```diff
@@ -68,22 +68,24 @@ define([
          * Allow navigating to an assessment via the URL.
          */
         _handleRoute: function(plugin, id) {
-            if (plugin === 'assessment' && typeof id !== 'undefined') {
-                // Check the 'id' passed is that of an article.
-                if (!Adapt.findById(id)) {
-                    // The 'id' passed may have been the assessment _id/name, not the article _id.
-                    var assessment = Adapt.assessment._assessments._byAssessmentId[id];
-                    if (assessment) {
-                        // Set 'id' to the article _id.
-                        id = assessment._id;
-                    } else {
-                        Adapt.log.warn('Assessment not found with _id: ' + id);
-                        return;
-                    }
-                }
+            if (plugin !== 'assessment' || id === undefined) {
+                return;
+            }
 
-                Backbone.history.navigate('#/' + id, { trigger: true, replace: true });
+            // Check the 'id' passed is that of an article.
+            if (!Adapt.findById(id)) {
+                // The 'id' passed may have been the assessment _id/name, not the article _id.
+                var assessment = Adapt.assessment._assessments._byAssessmentId[id];
+                if (assessment) {
+                    // Set 'id' to the article _id.
+                    id = assessment._id;
+                } else {
+                    Adapt.log.warn('Assessment not found with _id: ' + id);
+                    return;
```

Comment:
so, this code will error if the id isn't found on an assessment or in the course. it will otherwise route all ids as normal.
is that as intended or should it perhaps error if not going to an assessment id?

Reply:
It's as intended if the `id` doesn't belong to an article or an assessment, yes.  

I was thinking a `warn` rather than `error` as it's a circumstance the code is catching rather than an un-handled error that's being thrown, and it's consistent with the `warn` function in the router, e.g. if a user attempts to navigate to locked content.